src/main/java/net/minecraftforge/registries/NewRegistryEvent.java

```diff
@@ -62,8 +64,19 @@ void fill()
                 .collect(Collectors.toMap(RegistryData::builder, d -> d.builder().create(), (a, b) -> a, IdentityHashMap::new));
 
         builtRegistries.forEach((builder, reg) -> {
-            if (builder.getHasWrapper() && !Registry.REGISTRY.containsKey(reg.getRegistryName()))
+            RegistryAccess.RegistryData<?> dataPackRegistryData = builder.getDataPackRegistryData();
+            if (dataPackRegistryData != null) // if this is a datapack registry
+            {
+                if (builder.getHasWrapper() && !BuiltinRegistries.REGISTRY.containsKey(reg.getRegistryName()))
```

Comment:
Is it possible for a datapack registry to reach this part and have `getHasWrapper()` to be `false`? If not, should we remove the check since we 'know' that it is always `true`, move it to something like a `Validate` call (to indicate it should never be `false` but we check it anyway), or keep it as-is as a potential safeguard if _somehow_ something manages to make a datapack registry without a wrapper?

Reply:
Yeah, the registry builder automatically creates wrappers for all datapack registries, so we can remove the check